server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1402,6 +1403,11 @@ private boolean checkForHypervisorIdUpdate(Consumer existing, ConsumerDTO incomi
                 else {
                     // Safer to build a new clean HypervisorId object
                     Owner owner = ownerCurator.findOwnerById(existing.getOwnerId());
+                    if (isUsedByOwner(incomingId, owner)) {
```

Comment:
Can you help me understand under what condition this error is encountered?

Reply:
There are a few cases where consumer update/creation precondition is not validated before hitting the database which is causing the MySQL error. In this case consumer should not be able to update its hypervisor id to one already in use by owner. You can check the error with the following spec test: "should not allow a consumer to update their hypervisorId to one in use by owner".

Reply:
Ok, so to rephrase. We don't want to create two consumers in a single organization that represent the same hypervisor. In this particular case:
1) Two hypervisors consumers are created with different IDs
2) A change happens that causes the hypervisor IDs to converge.
3) Candlepin fails because it now has two records for what is effectively the same hypervisor. 

If the above is true, should we be merging the consumer records?

Reply:
Does it make sense to merge two hypervisors? ID conflict could be accidental.

Reply:
@kahowell ^

Reply:
I'm ok with this as is for now.

Reply:
> Does it make sense to merge two hypervisors? ID conflict could be accidental.

I think we need to design some process for merging them with a script, or provide customer guidance on how to handle failing requests when they've gotten into this state. I agree with throwing the bad request, as this is a state we are trying to avoid generally.